common/compositor/va/varenderer.cpp

```diff
@@ -488,29 +491,38 @@ bool VARenderer::UpdateCaps() {
   return true;
 }
 
-uint32_t VARenderer::HWCRotationToVA(uint32_t transform) {
-  switch (transform) {
-    case kTransform270:
-      return VA_ROTATION_270;
-    case kTransform180:
-      return VA_ROTATION_180;
-    case kTransform90:
-      return VA_ROTATION_90;
-    default:
-      break;
+#if VA_MAJOR_VERSION >= 1
+void VARenderer::HWCTransformToVA(uint32_t transform, uint32_t& rotation, uint32_t& mirror) {
+  rotation = VA_ROTATION_NONE;
+  mirror = VA_MIRROR_NONE;
+
+  if(!(transform & (kReflectX | kReflectY))) {
+    switch (transform) {
+      case kTransform270:
+        rotation = VA_ROTATION_270;
+      case kTransform180:
+        rotation = VA_ROTATION_180;
+      case kTransform90:
+        rotation = VA_ROTATION_90;
+      default:
+        break;
+    }
+  }
+  else {
+    if (transform & kReflectX)
```

Comment:
what happend if ((transform & kReflexX) && (transform & kReflexY)) is true?

thanks

Reply:
They can both be true, and both mirror can be possilbe, right, good findings